rcl/src/rcl/logging_rosout.c

```diff
@@ -74,6 +75,7 @@ typedef struct rosout_map_entry_t
 static rcutils_hash_map_t __logger_map;
 static bool __is_initialized = false;
 static rcl_allocator_t __rosout_allocator;
+static rcutils_array_list_t __sublogger_names;
```

Comment:
I'm wondering whether this should be a hash_map, rather than a list.  That way when removing them, we don't have to look through all subloggers to find it, which should improve performance.  Thoughts?